File: src/main/java/javafiles/dataaccessfiles/XMLIOBuilder.java
package javafiles.dataaccessfiles;

import javafiles.customexceptions.ReadWriteException;

class XMLIOBuilder implements FileIOReaderBuilder {
    private final String[] EXTENSIONS;

    XMLIOBuilder(String[] extensions) {
        EXTENSIONS = extensions;
    }

    @Override
    public FileIO createFileIOReader(String path) throws ReadWriteException {
        return new XMLIO(path, 'r');
    }

    @Override
    public String[] getExtensions() {
        return EXTENSIONS;
    }
}

File: src/main/java/javafiles/dataaccessfiles/builderimplements/XMLIOBuilder.kt
package javafiles.dataaccessfiles.builderimplements

import javafiles.dataaccessfiles.fileioimplements.FileIOReader
import javafiles.dataaccessfiles.fileioimplements.XMLIO
import java.io.File

internal class XMLIOBuilder(override val extensions: Array<String>) : FileIOReaderBuilder {
    override fun createFileIO(path: String): FileIOReader {
        return XMLIO(File(path))
    }
}
